src/TheThingsNetwork.cpp

```diff
@@ -16,6 +16,10 @@
 #define MAC_TX_OK "mac_tx_ok"
 #define MAC_RX "mac_rx"
 
+#define CHECK_CONFIGURATION "Check your coverage, keys and backend status."
+#define JOIN_INCORRECT_KEYS "Invalid keys: check AppKey and AppEui"
+#define PERSONALIZE_INCORRECT_KEYS "Invalid keys: check DevAddr, NwkSKey and AppSKey"
```

Comment:
Good idea to check for length, but the error messages are not clear. In this case you mean that the keys are of invalid length. Even if they are in of the correct length, then they might still be invalid. So maybe here use a single:

```c
#define ERROR_KEY_LENGTH "One or more keys are of invalid length."
```